src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_AssignmentOperator.cs

```diff
@@ -218,7 +218,7 @@ private BoundExpression MakeStaticAssignmentOperator(SyntaxNode syntax, BoundExp
 
                 case BoundKind.DiscardExpression:
                     {
-                        return rewrittenRight;
+                        return EnsureNotLvalueReceiver(rewrittenRight);
```

Comment:
yes, IDE did not rename all the usages though. Fixed now.

---
In reply to: [108249971](https://github.com/dotnet/roslyn/pull/18166#discussion_r108249971) [](ancestors = 108249971)